source/material_model/rheology/friction_options.cc

```diff
@@ -0,0 +1,849 @@
+
+        if (this->simulator_is_past_initialization() && this->get_timestep_number() > 0
+            && in.requests_property(MaterialProperties::reaction_terms)
+            && in.current_cell.state() == IteratorState::valid)
+          {
+            // q is from a for-loop through n_evaluation_points
+            const double current_edot_ii =
+              MaterialUtilities::compute_current_edot_ii (in.composition[q], ref_strain_rate,
+                                                          min_strain_rate, in.strain_rate[q],
+                                                          average_elastic_shear_moduli, use_elasticity,
+                                                          use_reference_strainrate, dte);
+
+            const double theta_old = in.composition[q][theta_composition_index];
+            double current_theta = 0;
+
+            for (unsigned int j=0; j < volume_fractions.size(); ++j)
+              {
+                const double critical_slip_distance = get_critical_slip_distance(in.position[q], j);
+                current_theta += volume_fractions[j] * compute_theta(theta_old, current_edot_ii, cellsize, critical_slip_distance);
+              }
+            const double theta_increment = current_theta - theta_old;
+
+            out.reaction_terms[q][theta_composition_index] = theta_increment;
+          }
+      }
+
+
+
+      template <int dim>
+      std::pair<double,double>
+      FrictionOptions<dim>::calculate_depth_dependent_a_and_b(const Point<dim> &position, const int j) const
+      {
+        Utilities::NaturalCoordinate<dim> point_a =
+          this->get_geometry_model().cartesian_to_other_coordinates(position, coordinate_system_a);
+        Utilities::NaturalCoordinate<dim> point_b =
+          this->get_geometry_model().cartesian_to_other_coordinates(position, coordinate_system_b);
+
+        const double rate_and_state_parameter_a =
+          rate_and_state_parameter_a_function->value(Utilities::convert_array_to_point<dim>(point_a.get_coordinates()),j);
+        const double rate_and_state_parameter_b =
+          rate_and_state_parameter_b_function->value(Utilities::convert_array_to_point<dim>(point_b.get_coordinates()),j);
+
+        return std::pair<double,double>(rate_and_state_parameter_a,
+                                        rate_and_state_parameter_b);
+      }
+
+
+
+      template <int dim>
+      double
+      FrictionOptions<dim>::get_critical_slip_distance(const Point<dim> &position, const int j) const
+      {
+        Utilities::NaturalCoordinate<dim> point =
+          this->get_geometry_model().cartesian_to_other_coordinates(position, coordinate_system_L);
+
+        const double critical_slip_distance =
+          critical_slip_distance_function->value(Utilities::convert_array_to_point<dim>(point.get_coordinates()),j);
+
+        if (critical_slip_distance <= 0)
+          AssertThrow(false, ExcMessage("Critical slip distance in a rate-and-state material must be > 0."));
```

Comment:
```suggestion
          AssertThrow(critical_slip_distance > 0, ExcMessage("Critical slip distance in a rate-and-state material must be > 0."));
```
The Asserts are designed to have the condition that should be satisfied inside them.

Reply:
Are there similar Asserts you can apply to a and b?